.maintain/gitlab/check_companion_build.sh

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bash
+#
+# check if a pr is compatible with companion pr or master if not available
+#
+# to override one that was just mentioned mark companion pr in the body of the
+# pr like
+#
+# $REPO companion: $ORGANISATION/$REPO#567
+#
+# $ORGANISATION and $REPO are set using $1 and $2. You can also specify a custom
+# build command with $3
+# Every argument after $3 is for specifying *additional* dependencies this
+# project has that depend on substrate, which might also have companion PRs.
+
+# Example: Cumulus relies on both substrate and polkadot. If this substrate PR
+# requires a companion build on polkadot, when we are testing that cumulus builds
+# with this commit of substrate, we *also* need to clone & patch polkadot, and tell
+# cumulus to use this cloned+patched version, else the build is guaranteed to fail
+# (since it doesn't have the changes to polkadot that were required in the polkadot
+# companion PR)
+
+# So invoke this script like (arguments in [] indicate optional arguments)
+# ./check_companion_build.sh paritytech cumulus ['cargo test --release'] [paritytech/polkadot]
+
+set -e
+
+# Include the common functions library
+#shellcheck source=../common/lib.sh
+. "$(dirname "${0}")/../common/lib.sh"
+
+ORGANISATION=$1
```

Comment:
nit: Apparently "Organisation" is valid spelling in British English but IMO it'd look better ORGANI**Z**ATION